### ISSUE TITLE
TextBlocks are not read by screen readers when they are used in a XAML Islands app

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/microsoft-ui-xaml#4399]
**Describe the bug**

`TextBlock`s are not read by Narrator and NVDA when they are used in a XAML Islands app.

**Steps to reproduce the bug**
Verified this behavior in the following places:
- Windows Terminal Settings UI (NavigationView scenario)
- [Xaml Islands Sample App](https://github.com/microsoft/Xaml-Islands-Samples)
  - "Win UI Navigation" is not read in the navigation view (even if a button is placed as the content next to the text block)
  - (when added) a text block above the "Hello from Xaml!" text box is not read

I have also tried adding a hyperlink to the text blocks, and the behavior slightly changed:
|  |Narrator|NVDA| 
|--|--|--|
|Normal| Reads hyperlink, but not surrounding text. |Reads text block and hyperlinked portion.|
|Xaml Islands| Silence. | Silence. |

If you enter scan mode, you are able to navigate around the text (hyperlinked and not).

**Expected behavior**

At a minimum, XAML Islands should have the same behavior as a non-xaml islands app.

Ideally, for hyperlinks, all screen readers would behave like NVDA. That said, that might be an issue that's on screen readers, not XAML.

The real issue here is that these text blocks are not read automatically. It seems like they're being exposed, and we're just not notifying the screen reader that we've navigated to a portion of the screen that has a text block.

**Version Info**
NuGet package version: 
[Microsoft.UI.Xaml 2.5.0]

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
|  |  |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| October 2020 Update (19042)        |    |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            |  |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**
None

**Comment by StephenLPeters** [comment]
@JesseCol or @Austin-Lamb can you weigh in?

**Comment by DHowett** [comment]
@saharsh979 from one employee here to another, who has some experience working in a public repo where the community can see the entire history of a bug:

Would you be willing to explain for the record why this accessibility bug is marked "Won't Fix"? Did we get an exception from the Accessibility team?

**Comment by Saharsh979** [comment]
Hi @DHowett You’re right to ask, and I apologize for the miscommunication. The GitHub action was performed automatically by our ADO-to-GitHub mirror as the state of the mirrored ADO bug got switched. I did not make the underlying “Won’t Fix” decision. I have not found a recorded Accessibility-team exception. I’m coordinating internally to get clarity and will update this thread with confirmed context.
